platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/UriNotationConverter.java

```diff
@@ -93,6 +93,6 @@ private static InvalidUserDataException invalidUserDataException(Object notation
     }
 
     private static boolean isWindowsRootDirectory(@Nullable String scheme) {
-        return scheme != null && scheme.length() == 2 && Character.isLetter(scheme.charAt(0)) && scheme.charAt(1) == ':' && OperatingSystem.current().isWindows();
+        return scheme != null && scheme.length() == 1 && Character.isLetter(scheme.charAt(0)) && OperatingSystem.current().isWindows();
```

Comment:
🤔 I wonder if we should take OS into account here. OTOH, the file parser will produce something weird for a windows-style path on other platforms anyway.

Reply:
Yep. I think it's better to fail earlier and these paths won't make much sense on non-windows OS anyway. Also, this "preserves" old behavior
https://github.com/gradle/gradle/blob/a3cacb207fec727859be9354c1937da2e59004c1/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/FileOrUriNotationConverter.java#L140